test/src/test/java/org/corfudb/integration/CorfuReplicationMultiSourceSinkIT.java

```diff
@@ -500,18 +491,40 @@ public ReplicatedStreamsListener(CountDownLatch countdownLatch, boolean snapshot
+            //  the replicated streams received from their own vs other sessions.  So if multiple writers find a
+            //  stream in the global list of streams to be replicated, each will apply updates from the shadow
+            //  stream of that stream, even though no data was received for it from the writer's own session.
+            //  In this test, Source-1 has data in Table-1, Source-2 in Table-2 and so on.  Each Source has a
+            //  separate session with the Sink.  Even though the session corresponding to Source-2 does not receive
+            //  data from Source-1(Table-1), the Sink writer will apply it because both Tables 1 and 2 are included
+            //  in the global streams to replicate set.
+            //  So the number of updates received on the Sink on a snapshot sync can be more than expected.
+            //  Hence, we perform the following modified check:
+            //  If Snapshot Sync:  Only consider the 1st set of updates(1st transaction) received for a given table.
+            //  However, this will lead to another test issue that subsequent async updates on the table will still
+            //  be received by this listener and interfere with the verification of log entry sync updates.
+            //  So we filter out these updates by looking at the number of entries in the transaction(1 entry = log
+            //  entry sync.  This assumption can work for the test as it writes single entries during log entry sync.)
+            //  Once the above limitation is addressed, these special checks must be removed.
             results.getEntries().forEach((schema, entries) -> {
-                if (snapshotSync) {
-                    if (entries.size() > 1) {
-                        processUpdate(schema, entries);
-                    }
-                } else {
+                // Ignore single 'clear' updates made to clear local writes on the Sink
+                if (snapshotSync && entries.size() > 1) {
+                    processUpdate(schema, entries);
+                } else if (!snapshotSync && entries.size() == 1) {
+                    // If in Log Entry sync, ignore updates with >1 entries (as they are from the redundant updates
+                    // received during snapshot sync), as explained in the TODO
                     processUpdate(schema, entries);
                 }
             });
         }
 
         private void processUpdate(TableSchema schema, List<CorfuStreamEntry> entries) {
+            // Ignore redundant updates received for a given table during snapshot sync, as explained in the above TODO
+            if (snapshotSync && tableToOpTypeMap.containsKey(schema.getTableName())) {
+                return;
+            }
+
             List<CorfuStreamEntry.OperationType> opList = tableToOpTypeMap.getOrDefault(
                 schema.getTableName(), new ArrayList<>());
             entries.forEach(entry -> opList.add(entry.getOperation()));
```

Comment:
Just for my understanding. Can you explain little further on this logic?

Reply:
discussed offline